cropper/src/main/java/com/canhub/cropper/BitmapUtils.kt

```diff
@@ -727,21 +714,25 @@ internal object BitmapUtils {
                         rect.width(), rect.height(), reqWidth, reqHeight
                     )
                 )
-            val stream = context.contentResolver.openInputStream(uri)
-            val decoder = BitmapRegionDecoder.newInstance(stream!!, false)
-            do {
+
+            context.contentResolver.openInputStream(uri).use {
```

Comment:
Is this stream closed automatically? I could not find something that guarantee it.

Reply:
Have you looked at the implementation of `use`?

Reply:
I didn't know the use of the keyword `use` either. I just looked it up. This is the given description of `use` in kotlin.

> Executes the given [block](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html#kotlin.io$use(kotlin.io.use.T,%20kotlin.Function1((kotlin.io.use.T,%20kotlin.io.use.R)))/block) function on this resource and then closes it down correctly whether an exception is thrown or not.

Links for more info
1. [Kotlin `use` documentation](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html#:~:text=Executes%20the%20given%20block%20function,exception%20is%20thrown%20or%20not.)
2. [A stackoverflow explanation of `use`](https://stackoverflow.com/a/39500046/4067774)